Library/Homebrew/os/mac/sdk.rb

```diff
@@ -93,10 +93,21 @@ def sdk_paths
 
             # Use unversioned SDK path on Big Sur to avoid issues such as:
             # https://github.com/Homebrew/homebrew-core/issues/67075
-            if OS::Mac.version >= :big_sur
-              sdk_path = File.join(sdk_prefix, "MacOSX.sdk")
-              version = OS::Mac.full_version
-              paths[version] = sdk_path if File.directory?(sdk_path)
+            sdk_path = File.join(sdk_prefix, "MacOSX.sdk")
+            if OS::Mac.version >= :big_sur && File.directory?(sdk_path)
+              sdk_settings_json = File.join(sdk_path, "SDKSettings.json")
+              sdk_settings_plist = File.join(sdk_path, "SDKSettings.plist")
+
+              if File.exist?(sdk_settings_json)
+                version = JSON.parse(File.read(sdk_settings_json))["Version"]
+              elsif File.exist?(sdk_settings_plist)
+                p = Utils.popen_read("plutil", "-convert", "json", "-o", "-", sdk_settings_plist)
```

Comment:
I think this might be excessive unless we've seen situations where one exists and the other doesn't.

Reply:
The JSON file was introduced in the 10.14 SDK I believe. SDKSettings.plist exists in all versions, even current ones for backwards compatibility reasons (Xcode itself switched to use JSON).

For plists: I think we already pull in a Ruby plist library that we can use rather than having to shell out.

Reply:
Given that it's Apple I think we can rely on the `plist` file to be there. Would you know if we can rely on the `json` file being there, @Bo98?

Reply:
Actually given this is `>= :big_sur`, I think we can rely on JSON only here.

Reply:
Excellent. JSON-only it is then.

Reply:
I agree we can skip this and use only JSON.